gears/src/x/ante.rs

```diff
@@ -142,6 +147,70 @@ impl<AK: AuthKeeper<SK>, BK: BankKeeper<SK>, SK: StoreKey, GC: SignGasConsumer>
         Ok(())
     }
 
+    fn mempool_fee<M: TxMessage, DB: Database>(
+        &self,
+        ctx: &mut TxContext<'_, DB, SK>,
+        TxWithRaw {
+            tx,
+            raw: _,
+            tx_len: _,
+        }: &TxWithRaw<M>,
+    ) -> anyhow::Result<()> {
+        if !ctx.is_check() {
+            return Ok(());
+        }
+
+        let fee = tx.auth_info.fee.amount.as_ref();
+        let gas = tx.auth_info.fee.gas_limit;
+
+        let min_gas_prices = ctx.options.min_gas_prices();
+
+        if min_gas_prices.is_empty() || min_gas_prices.iter().any(|this| this.amount.is_zero()) {
+            return Ok(());
+        }
+
+        if let Some(fee_coins) = fee {
+            let mut required_fees = Vec::with_capacity(min_gas_prices.len());
+
+            for gp in min_gas_prices {
+                required_fees.push(Coin {
+                    denom: gp.denom,
+                    amount: gp
+                        .amount
+                        .checked_mul(Decimal256::new(gas.into()))?
+                        .to_uint_ceil(),
+                });
+            }
+
+            if !is_any_gte(fee_coins.inner(), &required_fees) {
```

Comment:
Does this mean that if one fee is greater than the required fee then the tx passes?

Reply:
Right. This code from cosmos 

```go
if !feeCoins.IsAnyGTE(requiredFees) {
  return ctx, sdkerrors.Wrapf(sdkerrors.ErrInsufficientFee, "insufficient fees; got: %s required: %s", feeCoins, requiredFees)
}
```

Reply:
Ah I just realized this is the expected behaviour.